### DISCUSSION TITLE
ISR: cache never stale for fallback pages with a big revalidate time

### DISCUSSION THREAD

**Discussion by valse** [vercel/next.js#17944] [Help]
Hi, I'm trying to understand what is happening but even look at the source code I found an answer.

I'm in a single nodejs server, the site is on production with about 30k pages and I'm using the ISR feature for all of them: I made a lot of tests and I saw that pages with fallback enabled (true or blocking is the same) aren't never updated if their time is >= 60.

I have a test site too and in this low loaded process all works fine... so the problem happens with an high loaded nodejs process.

The strange things is that if I set the revalidate time to 10 seconds, the page is revalidated... maybe not exactly after 10 seconds but the regeneration process works for it.

To test I print a console log with a date created on the `getStaticProps` method and for the "immutable" pages this value still the same obviously...

I saw that under the hood you use an LRU cache of 50 MB for the incremental check... so if I set a bigger revalidate time maybe this cache increase until the old entries are deleted and not revalidated because out of cache?

Thanks!

**Comment by valse** [comment]
I saw also that revalidating pages without a fallback works as expected so I think that is related how the incremental cache is saved and I think by logical path (/blog/[slug]) and not by the real pathname... I'll try to investigate more.

**Reply by ivanpetrushev** [reply]
Hey, did you got any further investigating this?

We are having the same issue - large number of pages, 1800 seconds revalidate time - some of the pages are updated good, others are just stuck and never revalidated.

If we change revalidate to 60 seconds, it appears to solve the problem (for now).

**Reply by johannessteu** [reply]
We're running in the very same issue with a revalidate of `360`. We got and advent calendar up and running deployed in a kubernetes cluster with at least 3 application pods running the next-js project. At some point at least one pod is stuck and showing an outdated status because the pod never revalidated the page. We also added a date-string wich is generated on `getServerSideProps` that shows the last build time wich is in that case yesterday.

**Reply by ivanpetrushev** [reply]
Adding the generated date (and revalidate time!) to pages somewhere is really helpful investigating this. We add it in a html meta tag. 
We had one stupid issue getting revalidate to work at all. The revalidate time was being read from env file as a string while nextjs expects a number. Just leaving this here in case anyone else is banging his head on the wall as we did.
